src/Refactor/Xml/XmlWriter.php

```diff
@@ -0,0 +1,95 @@
+<?php
+namespace DemandwareXml\Refactor\Xml;
+
+use DemandwareXml\Refactor\Entity\WriteableEntityInteface;
+use SplFileObject;
+use XMLWriter as PhpXmlWriter;
+
+// Enhances XMLWriter with additional functionality and Demandware specific formatting.
+class XmlWriter extends PhpXmlWriter
+{
+    const NAMESPACE    = 'http://www.demandware.com/xml/impex/catalog/2006-10-31';
+    const INDENT_SPACE = ' ';
+
+    private $notEmptyWriter;
+
+    public function openFile(string $filename): bool
+    {
+        new SplFileObject($filename, 'w');
```

Comment:
I used `SplFileObject` here because it'll throw an exception if the file can't be created.